dotnet/src/Connectors/Connectors.GoogleVertexAI/GeminiPromptExecutionSettings.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+
+/// <summary>
+/// Represents the settings for executing a prompt with the Gemini model.
+/// </summary>
+[JsonNumberHandling(JsonNumberHandling.AllowReadingFromString)]
+public sealed class GeminiPromptExecutionSettings : PromptExecutionSettings
```

Comment:
I was going to comment here that there's no support for generically specifying a function call behavior via the base PromptExecutionSettings, and then I realized that was removed as part of https://github.com/matthewbolanos/semantic-kernel/commit/d32802e416e898e6e82a60b0ab13a1a5ca5839ea#diff-2806827066f872cf10cc8ac3d6ff1989abc4971d08a953b341f0698b8429f65aL75-L82. @markwallace-microsoft, was that intentional? This ties in with the work item to come up with a generalized function calling abstraction / mechanism.

Reply:
@stephentoub Sergey is currently working on defining the core abstractions to support this. We will have three implementations to base the design on (OpenAI, Gemini and MistralAI).